src/Aside/Aside.js

```diff
@@ -1,34 +1,53 @@
 import React, { Component } from 'react';
 import Controls from '../Controls/Controls.js';
 import AsideBody from '../AsideBody/AsideBody.js';
+import {cards} from '../mtg.js'
 import './Aside.css';
 
 class Aside extends Component {
   constructor() {
     super();
     this.state = {
-      cardNames: []
+      cardNames: [],
+      hasDuplicates: false,
```

Comment:
Is this the right spot for hasDuplicates? I'm wondering if it gets changed to true, won't it throw the duplicate error for every card? I could definitely be missing the way this is passed around.

Reply:
I think it hasDuplicates should be in Aside's state because the cardNames array is there. When addCardToList is called, a condition is checked to see if the array already has the cardName that was submitted. If is not in the array, the name is added to the list. If the name is in the array, an error message is shown instead of adding the name to the list.

When a different valid name is submitted, the error goes away. Also if an invalid name is submitted, the error for duplicates is removed and an error for invalid name is shown instead.